lib/std/mem/Allocator.zig

```diff
@@ -318,6 +318,13 @@ pub fn dupeZ(allocator: Allocator, comptime T: type, m: []const T) ![:0]T {
     return new_buf[0..m.len :0];
 }
 
+inline fn ensureSlice(comptime Type: type, comptime function_name: []const u8) std.builtin.Type.Pointer {
```

Comment:
An alternative idea here would be to accept a parameter like `src: std.builtin.SourceLocation`, so the calling function could just do `ensureSlice(T, @src())` - that could reduce future maintenance, although it's not super important, since it's not as though the API is bound to change that much.

Reply:
I'd tried this but it complained about the use in the type signature of `reallocAdvanced` as `@src()` is restricted to only be usable in a function body.
```zig
pub fn reallocAdvanced(
    self: Allocator,
    old_mem: anytype,
    new_n: usize,
    return_address: usize,
) t: {
    const Slice = ensureSlice(@TypeOf(old_mem), "reallocAdvanced"); // <-- Error here
    break :t Error![]align(Slice.alignment) Slice.child;
} {
    const Slice = ensureSlice(@TypeOf(old_mem), "reallocAdvanced"); // <-- Fine here
...
```

Maybe there's a better way of writing that function signature though? I agree that using `@src()` a nicer way of doing it, if possible.

Reply:
Right, then perhaps it would suffice to just do `ensureSlice(T, @src().fn_name)` where applicable, and otherwise do it manually where `@src` isn't allowed.